### ISSUE TITLE
Bug: [no-restricted-imports] `allowTypeImports` should allow inline type qualifiers

### ISSUE THREAD

**Issue by aaronadamsCA** [typescript-eslint/typescript-eslint#7374]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.5&showAST=es&fileType=.tsx&code=KYDwDg9gTgLgBAbzjAnmYcBiEJwL5wBmUEAtnAOQCWpksAtIThQNxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uhXyrPyIAJrUoBbYh2j5UGANrhsORNGhTIAGkVKsS7JGIBDfAAtZYBXr26rSyE0NjE6KOMnTaAMw4dN22ziG8PAcAO4AKiSIAJISUjIufLCI-lYAvqnYALqZGUo5EBlpQA&tsconfig=N4XyA&tokens=false

### Repro Code

```typescript
export { type Foo } from 'import-foo';
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-restricted-imports": [
      "error",
      {
        "paths": [
          {
            "name": "import-foo",
            "allowTypeImports": true
          }
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{}
```

### Expected Result

No error.

### Actual Result

`'import-foo' import is restricted from being used.`

**Comment by aaronadamsCA** [comment]
This might actually be a feature request, but it's _way_ easier to create a bug report using the playground, so you get a bug report. 😅

**Comment by bradzacher** [comment]
I think it makes sense to allow export from as long as every single specifier uses an inline type qualifier!
